### PR TITLE
Fix race condition in MySQL pool

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

race https://gist.github.com/alesapin/07552be387eb5d0e88e2a76b4a7adbb4

### CODE REVIEW

File: base/mysqlxx/Pool.h [hunk 1]
```diff
@@ -35,7 +36,9 @@ class Pool final
     struct Connection
     {
         mysqlxx::Connection conn;
-        int ref_count = 0;
+        /// Ref count modified in constructor/descructor of Entry
+        /// but also read in pool code.
+        std::atomic<int> ref_count = 0;
```
Comment (nikitamikhaylov) [inline base/mysqlxx/Pool.h:41]: Maybe also add relaxed memory order into all reads and writes?

Comment (alesapin) [reply base/mysqlxx/Pool.h:41]: But, it's a real race if I understand right? We can make the wrong decision using relaxed memory order? But of course, thread sanitizer will shut up even with relaxed memory order. Also, it's just a connection pool for MySQL, there shouldn't be a lot of contention on this `ref_count`?

Comment (nikitamikhaylov) [reply base/mysqlxx/Pool.h:41]: > We can make the wrong decision using relaxed memory order?
Maybe... Ok, it is too dangerous to play with it.

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by alesapin** [comment]
`test_multiple_disks/test.py::test_start_stop_moves` flaky test

**Pull request merged**